velox/row/UnsafeRowFast.cpp

```diff
@@ -0,0 +1,369 @@
+    case TypeKind::VARBINARY: {
+      auto value = decoded_.valueAt<StringView>(index);
+      return alignBytes(value.size());
+    }
+    case TypeKind::ARRAY:
+      return arrayRowSize(index);
+    case TypeKind::MAP:
+      return mapRowSize(index);
+    case TypeKind::ROW:
+      return rowRowSize(index);
+    default:
+      VELOX_UNREACHABLE();
+  };
+}
+
+bool UnsafeRowFast::isNullAt(vector_size_t index) {
+  return decoded_.isNullAt(index);
+}
+
+int32_t UnsafeRowFast::serialize(vector_size_t index, char* buffer) {
+  return serializeRow(index, buffer);
+}
+
+void UnsafeRowFast::serializeFixedWidth(vector_size_t index, char* buffer) {
+  switch (typeKind_) {
+    case TypeKind::BOOLEAN:
+      *reinterpret_cast<bool*>(buffer) = decoded_.valueAt<bool>(index);
+      break;
+    case TypeKind::TIMESTAMP:
+      *reinterpret_cast<int64_t*>(buffer) =
+          decoded_.valueAt<Timestamp>(index).toMicros();
+      break;
+    default:
+      memcpy(
+          buffer,
+          decoded_.data<char>() + decoded_.index(index) * valueBytes_,
+          valueBytes_);
+  }
+}
+
+void UnsafeRowFast::serializeFixedWidth(
+    vector_size_t offset,
+    vector_size_t size,
+    char* buffer) {
+  assert(supportsBulkCopy_);
+  memcpy(
+      buffer,
+      decoded_.data<char>() + decoded_.index(offset) * valueBytes_,
+      valueBytes_ * size);
+}
+
+int32_t UnsafeRowFast::serializeVariableWidth(
+    vector_size_t index,
+    char* buffer) {
+  switch (typeKind_) {
+    case TypeKind::VARCHAR:
+    case TypeKind::VARBINARY: {
+      auto value = decoded_.valueAt<StringView>(index);
+      memcpy(buffer, value.data(), value.size());
+      return value.size();
+    }
+    case TypeKind::ARRAY:
+      return serializeArray(index, buffer);
+    case TypeKind::MAP:
+      return serializeMap(index, buffer);
+    case TypeKind::ROW:
+      return serializeRow(index, buffer);
+    default:
+      VELOX_UNREACHABLE();
+  };
+}
+
+int32_t UnsafeRowFast::arrayRowSize(vector_size_t index) {
+  auto baseIndex = decoded_.index(index);
+
+  // array size | null bits | fixed-width data | variable-width data
+  auto arrayBase = decoded_.base()->asUnchecked<ArrayVector>();
+  auto offset = arrayBase->offsetAt(baseIndex);
+  auto size = arrayBase->sizeAt(baseIndex);
+
+  return arrayRowSize(children_[0], offset, size, fixedWidth_[0]);
+}
+
+int32_t UnsafeRowFast::serializeArray(vector_size_t index, char* buffer) {
+  auto baseIndex = decoded_.index(index);
+
+  // array size | null bits | fixed-width data | variable-width data
+  auto arrayBase = decoded_.base()->asUnchecked<ArrayVector>();
+  auto offset = arrayBase->offsetAt(baseIndex);
+  auto size = arrayBase->sizeAt(baseIndex);
+
+  return serializeAsArray(children_[0], offset, size, fixedWidth_[0], buffer);
+}
+
+int32_t UnsafeRowFast::mapRowSize(vector_size_t index) {
+  auto baseIndex = decoded_.index(index);
+
+  //  size of serialized keys array in bytes | <keys array> | <values array>
```

Comment:
```
nit: // The size of 
```

Reply:
I'd like to keep this as is since it is not a full sentence on purpose.